### PR TITLE
enh: Add snap-in with external GitHub Integration

### PR DESCRIPTION
**Pull request by altafDevRev**
## Summary

Adding the example snap-in that gets triggered from [GitHub push events](https://docs.github.com/en/webhooks/webhook-events-and-payloads#push) and makes a timeline entry using DevRev SDK. 

## Work-item
https://app.devrev.ai/devrev/works/ISS-70829

### CODE REVIEW

File: 7-external-github-webhook/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+## DevRev Snaps TypeScript Template
```
Comment (Atul-Butola) [inline 7-external-github-webhook/README.md:1]: Should it be snap-in or snaps?

File: 7-external-github-webhook/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+This repository contains a template for the functions that can be deployed as
```
Comment (Atul-Butola) [inline 7-external-github-webhook/README.md:3]:
```suggestion
This repository contains a template for functions that can be deployed as
```

File: 7-external-github-webhook/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+part of Snap-Ins.
```
Comment (Atul-Butola) [inline 7-external-github-webhook/README.md:4]:
```suggestion
part of snap-ins.
```

File: 7-external-github-webhook/README.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+For more reference on Snap-Ins, please refer to the [documentation](https://github.com/devrev/snap-in-docs)
```
Comment (Atul-Butola) [inline 7-external-github-webhook/README.md:6]:
```suggestion
For more information on snap-ins, refer to the [documentation](https://github.com/devrev/snap-in-docs)
```

File: 7-external-github-webhook/README.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+### Getting started with the template
+1. Create a new repository from this template.
+2. In the new repository, you can add functions at path `src/functions` where the folder name corresponds to the function name in your manifest file.
```
Comment (Atul-Butola) [inline 7-external-github-webhook/README.md:10]:
```suggestion
2. In the new repository, add functions at path `src/functions` where the folder name corresponds to the function name in your manifest file.
```

File: 7-external-github-webhook/README.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+3. Each function you add will also need to be mentioned in `src/function-factory.ts` .
```
Comment (Atul-Butola) [inline 7-external-github-webhook/README.md:11]:
```suggestion
3. Each function you add must also be mentioned in `src/function-factory.ts` .
```

File: 7-external-github-webhook/README.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+You can test your code by adding test events under `src/fixtures` similar to the example event provided. You can add keyring values to the event payload to test API calls as well.
+
+Once you have added the event, you can test your code by running:
+```
+npm install
+npm run start -- --functionName=on_work_creation --fixturePath=on_work_created_event.json
+```
+
+### Adding external dependencies
+You can also add dependencies on external packages in package.json under the “dependencies” key. These dependencies will be made available to your function at runtime and testing.
```
Comment (Atul-Butola) [inline 7-external-github-webhook/README.md:23]:
```suggestion
You can add dependencies on external packages in package.json under the “dependencies” key. These dependencies will be made available to your function at runtime and testing.
```

File: 7-external-github-webhook/README.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+npm run lint
+```
+
+To automatically fix fixable lint errors, run:
+
+```bash
+npm run lint:fix
+```
+
+### Deploying Snap-Ins
```
Comment (Atul-Butola) [inline 7-external-github-webhook/README.md:39]:
```suggestion
### Deploying snap-ins
```

File: 7-external-github-webhook/README.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+Once you are done with the testing, run the following commands to deploy your snap_in:
```
Comment (Atul-Butola) [inline 7-external-github-webhook/README.md:40]:
```suggestion
Once you are done with the testing, run the following commands to deploy your snap-in:
```

**Review by Atul-Butola — CHANGES_REQUESTED** [review]

File: 7-external-github-webhook/README.md (hunk 1 above)
Comment (navneel99) [reply 7-external-github-webhook/README.md:1]: Please update the README @altafDevRev

**Review by shashankcube — APPROVED** [review]

File: 8-external-github-webhook/code/.gitignore [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+# System Files
+.DS_Store
+Thumbs.db
+
+deps.json
+results.json
+before-run-data.json 
+
+build.tar.gz
```
Comment (kkviks) [inline 8-external-github-webhook/code/.gitignore:38]: nit: Can add new line char

File: 8-external-github-webhook/code/.eslintrc.js [hunk 11]
```diff
@@ -0,0 +1,48 @@
+module.exports = {
+  extends: 'airbnb-typescript/base',
+  extends: [
```
[the remark this answers is not in the document]
Comment (altafDevRev) [reply 8-external-github-webhook/code/.eslintrc.js:3]: Adding this in the same extends key.

File: 8-external-github-webhook/code/src/functions/github_handler/index.ts [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+    endpoint: endpoint,
+    token: token,
+  });
+
+  // Extract the part ID and commits from the event
+  const partID = event.input_data.global_values.part_id;
+  const commits = event.payload.commits;
+
+  // Iterate through commits and append the commit message to the body of the comment
+  let bodyComment = '';
```
Comment (coder336) [inline 8-external-github-webhook/code/src/functions/github_handler/index.ts:20]: Maybe add commiter and repo details at the start of the message.

File: 8-external-github-webhook/README.md [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+```
+devrev snap_in draft
+```
+4. Update the snap_in
+```
+devrev snap_in update
+```
+5. Deploy the snap_in
+```
+devrev snap_in deploy
```
Comment (coder336) [inline 8-external-github-webhook/README.md:52]:
```suggestion
devrev snap_in activate
```

File: 8-external-github-webhook/README.md [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+```
+3. Draft the snap_in
+```
+devrev snap_in draft
+```
+4. Update the snap_in
+```
+devrev snap_in update
+```
+5. Deploy the snap_in
```
Comment (coder336) [inline 8-external-github-webhook/README.md:50]:
```suggestion
5. Activate the snap_in
```

File: 8-external-github-webhook/README.md [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+```
+
+To automatically fix fixable lint errors, run:
+
+```bash
+npm run lint:fix
+```
+
+### Deploying snap-ins
+Once you are done with the testing, run the following commands to deploy your snap-in:
```
Comment (coder336) [inline 8-external-github-webhook/README.md:32]:
```suggestion
Once you are done with the testing, run the following commands to activate your snap-in:
```

File: 8-external-github-webhook/code/src/functions/github_handler/index.ts (hunk 12 above)
Comment (altafDevRev) [reply 8-external-github-webhook/code/src/functions/github_handler/index.ts:20]: Will add - `Commits from GitHub:\n`

**Pull request merged**